### PR TITLE
Add a migration guide for Noda Time 3.0.

### PR DESCRIPTION
**Pull request by malcolmr**
Also update the installation page with updated system requirements.

### CODE REVIEW

File: src/NodaTime.Web/Markdown/unstable/installation.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+[project download page][downloads].
+
+[downloads]: /downloads/
+
+See the ["Building and testing"][building] section in the developer guide for
+instructions on building Noda Time from source.
+
+[building]: /developer/building
+
+System requirements
```
Comment (malcolmr) [inline src/NodaTime.Web/Markdown/unstable/installation.md:20]: This is the only section I've changed here. (Is there some way to force a diff against `src/NodaTime.Web/Markdown/2.0.x/installation.md`?)

**Comment by malcolmr** [comment]
I have some additional questions and comments:

- Is there any way that marking structs `readonly` could be a breaking change? https://blog.paranoidcoding.com/2019/03/27/readonly-struct-breaking-change.html suggests not, though note that we did also introduce new `readonly` fields in some cases (e.g. for `Instant.duration`).
- Could the addition of the `System.Runtime.CompilerServices.Unsafe` dependency break anything? It seems like it's part of .NET Core, so I assume not, but are there deployment scenarios in which it might not be available such that you couldn't just drop in a new `NodaTime.dll`?
- Could removing `AllowPartiallyTrustedCallers` have any runtime implications?
  https://docs.microsoft.com/en-us/dotnet/api/system.security.allowpartiallytrustedcallersattribute says "Partially trusted code is no longer supported. This attribute has no effect in .NET Core.", but it's not obvious to me whether there could still be an impact under the **.NET Framework** runtime.
- `AnnualDate` doesn't implement `IXMLSerializable` in Noda Time 2.x. From what I can see, that means that attempting to serialize it would just produce a useless `<AnnualDate/>` output rather than anything more useful (e.g. it doesn't implicitly serialize the underlying fields), so adding support can't possibly be a breaking change(?)

Some other things I spotted along the way:

- `IDateTimeZoneProvider` should probably not talk about `BclDateTimeZone` being unavailable for netstandard1.3 any more(?).
- https://github.com/nodatime/nodatime.serialization/blob/master/README.md still says we support binary serialization.

File: src/NodaTime.Web/Markdown/unstable/installation.md (hunk 1 above)
Comment (jskeet) [reply src/NodaTime.Web/Markdown/unstable/installation.md:20]: Within GitHub? Not that I'm aware of. I could do it locally, but I trust you :)

**Comment by malcolmr** [comment]
(Any comments on the questions above?)

**Comment by jskeet** [comment]
Sorry, failed to see the questions.

- No, structs becoming readonly should not be breaking.
- I don't believe the addition of the unsafe package should break anything. Never say never when it comes to dependencies, but I think we're okay.
- Ooh, removing AllowPartiallyTrustedCallers *could* potentially break things, but frankly I find all of APTA confusing anyway. Worth calling out, yes.
- AnnualDate an XML serialization: well it's certainly a *change*, but I think if any code is relying on a broken representation, that's okay to break.

Good point about the other updates to make.

**Pull request merged**